staging/src/k8s.io/apiserver/pkg/endpoints/request/webhook_duration_test.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package request
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+func TestWebhookDurationFrom(t *testing.T) {
+	t.Run("TestWebhookDurationFrom", func(t *testing.T) {
+		const durSize = 5
+		durations := [durSize]time.Duration{}
+		sumDur := time.Duration(0)
+		for i := 0; i < durSize; i++ {
```

Comment:
Instead of generating this - please make the test static - define explicitly the parameters, sum and max and pass it down.
Something like:

```
type testCase struct {
  durations []time.Duration
  sumDurations time.Duration
  maxDuration  time.Duration
}{
  {
     durations: []time.Duration{100*time.Millisecond, ...},
     sumDuration: ...
...
```

Reply:
[Having tests that are explicit means that we're avoiding risk of test not doing what we believe it wasn't doing

Reply:
I agree, fixed.